install.sh

```diff
@@ -321,19 +328,30 @@ install() {
 
   determine_chipset
 
-  # captured in variable so no way of doing ! cmd
-  # shellcheck disable=SC2181
-  if ! determine_download_url; then
-    panic "Could not find download for release specified."
-  fi
+  if [ -z "${THIS_DIR}" ]; then
+    # captured in variable so no way of doing ! cmd
+    # shellcheck disable=SC2181
+    if ! determine_download_url; then
+      panic "Could not find download for release specified."
+    fi
 
-  # Download tarball
-  DOWNLOAD_URL="$(cat "${temp_d}"/download_url)"
+    # Download tarball
+    DOWNLOAD_URL="$(cat "${temp_d}"/download_url)"
 
-  log INFO 'Downloading CLI tarball. Will take a few moments...'
+    log INFO 'Downloading CLI tarball. Will take a few moments...'
 
-  if ! download_cli_tarball; then
-    panic "Could not download release tarball."
+    if ! download_cli_tarball; then
+      panic "Could not download release tarball."
+    fi
+  else
+    log 'DEBUG' 'Looking for tar to install in built artefacts'
+    for tar_ball in dist/*.tar.gz; do
+      log 'DEBUG' "tar_ball=${tar_ball}"
+      if printf -- '%s' "${tar_ball}" | grep -Eq "${OS}-${CHIPSET}\.tar\.gz$"; then
+        log 'DEBUG' "using tar_ball=${tar_ball}"
+        downloaded_cli_tar_file="${tar_ball}"
+      fi
+    done
```

Comment:
Should there be a check after done to see if anything has been assigned, or at least to log it?